spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/JobStepExecutionController.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.controller;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.batch.admin.service.JobService;
+import org.springframework.batch.admin.service.NoSuchStepExecutionException;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.launch.NoSuchJobExecutionException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.dataflow.rest.resource.StepExecutionResource;
+import org.springframework.hateoas.ExposesResourceFor;
+import org.springframework.hateoas.mvc.ResourceAssemblerSupport;
+import org.springframework.http.HttpStatus;
+import org.springframework.util.Assert;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author Glenn Renfro
+ */
+@RestController
+@RequestMapping("/jobs/executions/{jobExecutionId}/steps")
+@ExposesResourceFor(StepExecutionResource.class)
+public class JobStepExecutionController {
+
+	private final JobService jobService;
+
+	private final Assembler stepAssembler = new Assembler();
+
+	/**
+	 * Creates a {@code JobStepExecutionsController} that retrieves Job Step Execution
+	 * information from a the {@link JobService}
+	 *
+	 * @param jobService the service this controller will use for retrieving
+	 *  job execution information.
+	 */
+	@Autowired
+	public JobStepExecutionController(JobService jobService) {
+		Assert.notNull(jobService, "repository must not be null");
+		this.jobService = jobService;
+	}
+
+	/**
+	 * List all step executions.
+	 *
+	 * @param id the {@link JobExecution}.
+	 * @return Collection of {@link StepExecutionResource} for the given jobExecutionId
+	 */
+	@RequestMapping(value = { "" }, method = RequestMethod.GET)
+	@ResponseStatus(HttpStatus.OK)
+	public List<StepExecutionResource> stepExecutions(
+			@PathVariable("jobExecutionId") Long id) throws NoSuchJobExecutionException{
```

Comment:
use small `long` everywhere. Did you merge https://github.com/ericbottard/spring-cloud-dataflow/commit/9916016fdae764d3952fc897a510b3fd0cb1e913 ?